libgnucash/engine/guid.cpp

```diff
@@ -266,15 +256,9 @@ guid_hash_to_guint (gconstpointer ptr)
         PERR ("received nullptr guid pointer.");
         return 0;
     }
-    GncGUID const & guid = * reinterpret_cast <GncGUID const *> (ptr);
-    gnc::GUID const & temp {guid};
-
-    guint hash {0};
-    std::for_each (temp.begin (), temp.end (), [&hash] (unsigned char a) {
-        hash <<=4;
-        hash |= a;
-    });
-    return hash;
+    const GncGUID* g = static_cast<const GncGUID*>(ptr);
+    const std::string_view data{reinterpret_cast<const char*>(g->reserved), GUID_DATA_SIZE};
+    return std::hash<std::string_view>{}(data);
```

Comment:
`std::hash` produces a `size_t` that fits in an `gint` only on 32-bit systems. You need to change the return type here and in all uses.

Reply:
Also it would be better to make a gnc:GUID member function and wrap it for use in gnc-sx-instance-model.c and call it directly in import-pending-matches.cpp.